src/diffusers/models/resnet.py

```diff
@@ -465,9 +472,18 @@ def forward(self, input_tensor, temb):
 
         if temb is not None:
             temb = self.time_emb_proj(self.nonlinearity(temb))[:, :, None, None]
-            hidden_states = hidden_states + temb
 
-        hidden_states = self.norm2(hidden_states)
+            if self.time_embedding_norm == "default":
+                hidden_states = hidden_states + temb
+                hidden_states = self.norm2(hidden_states)
```

Comment:
could we move this out of the if-else cascade and just do it after -> then we would have just one simple if else statement:

if == "default":
    hidden_states = ...
elif == "scale_shift":
    scale, shift = ...
    hidden_states = ...
```

Reply:
yep!

Reply:
Ah sorry I forgot I had tried to do it like that but the scale and shift are applied after the norm is applied so I don't know if that's feasible? Lmk if I'm missing something

Reply:
Alternatively, we could do

```python
        if temb is not None:
            temb = self.time_emb_proj(self.nonlinearity(temb))[:, :, None, None]

            if self.time_embedding_norm == "default":
                hidden_states = hidden_states + temb
            elif self.time_embedding_norm == "scale_shift":
                scale, shift = torch.chunk(temb, 2, dim=1)
            else:
                raise ValueError(f"unknown time embedding norm config: {self.time_embedding_norm} ")
        
        hidden_states = self.norm2(hidden_states)

        if self.time_embedding_norm == "scale_shift":
            hidden_states = hidden_states * (1 + scale) + shift
```

To me, this is a little less clear than the current way because time_embedding_norm is checked for scale shift in two places. 

I don't have terribly strong opinions here though so open to anything :)

Reply:
Actually, I think this makes the most sense 

```python
        if temb is not None and self.time_embedding_norm == "default":
            temb = self.time_emb_proj(self.nonlinearity(temb))[:, :, None, None]
            hidden_states = hidden_states + temb

        hidden_states = self.norm2(hidden_states)

        if temb is not None and self.time_embedding_norm == "scale_shift":
            scale, shift = torch.chunk(temb, 2, dim=1)
            hidden_states = hidden_states * (1 + scale) + shift

```

I committed this way into the branch but will wait to resolve comment until confirmed